kube/compose.go

```diff
@@ -287,9 +350,28 @@ func (s *composeService) Remove(ctx context.Context, project *types.Project, opt
 
 // Exec executes a command in a running service container
 func (s *composeService) Exec(ctx context.Context, project *types.Project, opts api.RunOptions) (int, error) {
+	if err := checkUnsupportedExecOptions(opts); err != nil {
+		return 0, err
+	}
 	return 0, s.client.Exec(ctx, project.Name, opts)
 }
 
+func checkUnsupportedExecOptions(o api.RunOptions) error {
+	var errs *multierror.Error
+	checks := []struct {
+		toCheck, expected interface{}
+		option            string
+	}{
+		{o.Index, 0, "index"},
+		{o.Privileged, false, "privileged"},
+		{o.WorkingDir, "", "workdir"},
+	}
+	for _, c := range checks {
+		errs = utils.CheckUnsupported(errs, c.toCheck, c.expected, "exec", c.option)
+	}
```

Comment:
nit: honestly this wouldn’t be worse, would it?
```suggestion
	errs = utils.CheckUnsupported(errs, o.Index, 0, "exec", "index")
	errs = utils.CheckUnsupported(errs, o.Privileged, false, "exec", "privileged")
	errs = utils.CheckUnsupported(errs, o.WorkingDir, "", "exec", "workdir")
```

Reply:
Done